test/base/src/androidTest/kotlin/io/realm/kotlin/test/shared/RealmSetTests.kt

```diff
@@ -0,0 +1,1156 @@
+            Short::class to RealmSetContainer::shortSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any>>,
+            Int::class to RealmSetContainer::intSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any>>,
+            Long::class to RealmSetContainer::longSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any>>,
+            Boolean::class to RealmSetContainer::booleanSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any>>,
+            Float::class to RealmSetContainer::floatSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any>>,
+            Double::class to RealmSetContainer::doubleSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any>>,
+            RealmInstant::class to RealmSetContainer::timestampSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any>>,
+            ObjectId::class to RealmSetContainer::objectIdSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any>>,
+            ByteArray::class to RealmSetContainer::binarySetField as KMutableProperty1<RealmSetContainer, RealmSet<Any>>,
+            RealmObject::class to RealmSetContainer::objectSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any>>
+        ).toMap()
+
+        @Suppress("UNCHECKED_CAST")
+        val nullableProperties = listOf(
+            String::class to RealmSetContainer::nullableStringSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            Byte::class to RealmSetContainer::nullableByteSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            Char::class to RealmSetContainer::nullableCharSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            Short::class to RealmSetContainer::nullableShortSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            Int::class to RealmSetContainer::nullableIntSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            Long::class to RealmSetContainer::nullableLongSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            Boolean::class to RealmSetContainer::nullableBooleanSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            Float::class to RealmSetContainer::nullableFloatSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            Double::class to RealmSetContainer::nullableDoubleSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            RealmInstant::class to RealmSetContainer::nullableTimestampSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            ObjectId::class to RealmSetContainer::nullableObjectIdSetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>,
+            ByteArray::class to RealmSetContainer::nullableBinarySetField as KMutableProperty1<RealmSetContainer, RealmSet<Any?>>
+        ).toMap()
+    }
+}
+
+// Circular dependencies with sets
+class SetLevel1 : RealmObject {
```

Comment:
Is it necessary to have three different classes here? Couldn't we have just one?

Reply:
It follows the pattern we have in lists so I would keep it for consistency.